Packs/RetrievePlaybooksAndIntegrations/Scripts/RetrievePlaybooksAndIntegrations/RetrievePlaybooksAndIntegrations.py

```diff
@@ -0,0 +1,111 @@
+import traceback
+from typing import Any, Dict
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+''' STANDALONE FUNCTION '''
+
+
+def perform_rest_call(method: str, endpoint: str, body=None):
+    """
+    Perform a REST Call
+    :param endpoint:
+    :param body: `dict` representing the HTTP body
+    :param method: `get` or `post`
+    :return: The contents from the API Call
+    """
+    args = {
+        'uri': endpoint,
+    }
+    if body:
+        args['body'] = json.dumps(body)
+
+    result = demisto.executeCommand(f"demisto-api-{method}", args)
+
+    if len(result) < 1 or 'Contents' not in result[0] or 'response' not in result[0].get('Contents'):
+        raise Exception(f"Error with REST call to endpoint {endpoint}")
+
+    return result[0]['Contents']['response']
+
+
+def get_tasks_list(tasks):
+    # {'1':{'id': '1'}, ...} -> [{'id':'1'}, ...]
+    tasks_list = []
+    for task_number in tasks.keys():
+        tasks_list.append(tasks[task_number])
+    return tasks_list
+
+
+def append_to_playbooks_and_integrations(playbook_name, brands):
+    if playbook_name not in playbooks:
+        playbooks.append(playbook_name)
+    for b in brands:
+        if b and b != 'Builtin' and b not in integrations:
+            integrations.append(b)
+
+
+def get_subplaybook_tasks(task):
+    # recursively go through all subplaybook tasks and append to playbooks and integrations
+
+    subplaybook_json = perform_rest_call('get', f"playbook/{task['task']['playbookId']}")
+    append_to_playbooks_and_integrations(subplaybook_json['name'], subplaybook_json['brands'])
+    tasks = get_tasks_list(subplaybook_json['tasks'])
+    for t in tasks:
+        if t['type'] == 'playbook' and t['task'].get('playbookId'):
+            # playbookId does not exist if the playbook the task references is missing
+            get_subplaybook_tasks(t)
+
+
+''' COMMAND FUNCTION '''
+
+
+def _main(args: Dict[str, Any]) -> CommandResults:
+    query = f'''name:"{args['playbook_name']}"'''
+    body = {
+        'query': query
+    }
+    playbooks_json = perform_rest_call('post', 'playbook/search', body)
+    for playbook_json in playbooks_json['playbooks']:
+        if playbook_json['name'] == args['playbook_name']:
+            break
+    append_to_playbooks_and_integrations(playbook_json['name'], playbook_json['brands'])
+
+    tasks = get_tasks_list(playbook_json['tasks'])
+    for task in tasks:
+        if task['type'] == 'playbook':
+            get_subplaybook_tasks(task)
```

Comment:
which lines into which for loop?

Reply:
all these marked lines should be under the for loop above them, don't they?
this loop: `for playbook_json in playbooks_json['playbooks']:`.

Reply:
I don't believe so. lines 64-71 are for searching for the playbook by name which ultimately sets `playbooks_json` which is used in line 74 to set `tasks` which lines 75-77 iterate through

Reply:
So i'm not sure I understand the logic.
1. How do you call **playbook_json** if it exist only inside the for loop?
2. What's the idea of the for loop? you just break in case you find something and you don't do anything else inside it?

Reply:
But `playbook_json` does not exist only inside the for loop. The reason for the `for` loop on line 72 is to set `playbook_json` for the playbook name defined in the `playbook_name` input. `playbooks_json['playbooks']` can return a list of playbook jsons/dicts, so we need to loop through it and locate the one which has a playbook name (`playbook_json['name']`) of `playbook_name`

Reply:
OH, ok.. now I got it! my bad, sorry!